modelservice/src/main/java/com/expedia/adaptivealerting/modelservice/web/DetectorMappingController.java

```diff
@@ -73,9 +86,13 @@ public void deleteDetectorMapping(@RequestParam String id) {
     }
 
     @RequestMapping(value = "/search", method = RequestMethod.POST)
-    public List<DetectorMapping> searchDetectorMapping(@RequestBody SearchMappingsRequest request) {
+    public List<DetectorMapping> searchDetectorMapping(@RequestBody SearchMappingsRequest request,
+                                                       @RequestHeader HttpHeaders httpHeaders) {
+        SpanContext parentSpanContext = trace.extractParentSpan(httpHeaders);
+        Span span = trace.startSpan("mappings-search", parentSpanContext);
         AssertUtil.isTrue(request.getUserId() != null || request.getDetectorUuid() != null,
                 "User id and Detector UUID can't both be null");
+        span.finish();
```

Comment:
same as the previous comment, finishing the span should be after `search` method in order to get the correct duration/end-time of the span in Haystack

Reply:
Fixed.